src/ngraph_encapsulate_op.cc

```diff
@@ -195,6 +207,16 @@ class NGraphEncapsulateOp : public tf::OpKernel {
       }
       last_used_src_ptrs[i] = src_ptr;
       ng_inputs.push_back(t);
+      };
+
+      if(env_value == "NNP") {
+      auto t = m_ng_backend->create_tensor(ng_element_type, ng_shape);
```

Comment:
I'm not sure if GPU needs something similar to this.  @fengleitian can confirm.

Reply:
This needs to change. We shouldn't rely on env variable to determine the backend everywhere - only once as it's done during the initialization time. In this class we should get this from the context.

Reply:
Should be same. But let me double check.

Reply:
Additional step needed for GPU. check this:
https://github.com/NervanaSystems/ngraph-tf/pull/74/files

Reply:
@fengleitian Looks like I'm going to remove support for GPU as requested by @avijit-nervana  in this PR so I will let you take care of it.

Reply:
GPU works same as NNP. 
1. Add data copy from CPU side to Device side when create input tensor.
2. Add data copy from Device to CPU side after get result.